openapi/openapi.yaml

```diff
@@ -1287,16 +1292,31 @@ components:
           type: string
       required: [title, url]
 
-
-    User:
+    UserMetadataInDB:
       type: object
       properties:
-        email:
+        id:
+          description: |
+            identical to the sub in Cognito, which is a unique identifier for
+            the authenticated user.
           type: string
         preferred_username:
+          description: identical to the preferred_username in Cognito.
           type: string
-        username:
-          type: string
-        api_key:
-          type: string
-      required: [email, preferred_username, username, api_key]
+      required: [id, preferred_username]
+
+    UserMetadata:
+      allOf:
+        - type: object
+        - properties:
+            username:
```

Comment:
If `username` and `id` are the same things, do you think we can remove `username` from the public interface here?

It looks like we are always using `user.id` as the UID. I think maybe we can remove `username`s from the codebase entirely so it is clear that whenever we need a UID, we should always use `user.id`. I don't have a strong opinion on the second part though.

Reply:
Thanks! I thought about the same thing, it makes sense in favor of simplicity.

Reply:
I think removing it from the public interface makes sense!

For the second part, the Cognito function `client.admin_update_user_attributes()` only accepts `username`, so I preserved it in the codebase. While we can pass in id since it's the same as username in our case, I think it's still better to avoid mixing them?